tools/devtool

```diff
@@ -375,6 +393,110 @@ run_devctr() {
+# to install and add freeze their version in {arch-based}-requirements_freeze.txt
+# 
+# Args:
+#   $1  path to the {arch-based}-requirements-freeze file.
+# 
+update_packages() {
+    declare requirements_freeze_file_location="$1"  
+    declare parsed_package_location=$(mktemp)
+    say "Checking for newly added requirements..."
+    while IFS= read -r line; do
+        echo "${line%%=*}" >>$parsed_package_location
+    done <"$requirements_freeze_file_location"
+    unset IFS
+
+    diff=($(grep -v -f $parsed_package_location $FC_DEVCTR_DIR/requirements/requirements.txt))    
+    rm $parsed_package_location
+
+    if [ "${#diff[@]}" -ne 0 ]; then
+        say "List of newly added packages"
+        printf '%s\n' "${diff[@]}"
+        say "Would you like to install and freeze versions of these packages?"
+        select yn in "Yes" "No"; do
+            case $yn in
+                Yes ) perform_update $requirements_freeze_file_location ${diff[@]}; break;;
+                No ) exit;;
+            esac
+        done 
+    else
+       say "Eveything up-to-date..."
+    fi
+}
+
+# Given an image-id, launches a new docker container.
+# It then runs a version upgrade of outdated packages on container, freezes
+# the new version to a requirements file and copies it back to the host.
+# 
+# Args:
+#   $1  image id 
+#   $2  path to the {arch-based}-requirements-freeze file.
+# 
+perform_upgrade() {
+    declare image_id="$1"
+    declare requirements_freeze_file_location="$2"
+
+    dummy_container_name=$(uuidgen)
+    dummy_container_id=$(docker create -ti --name "$dummy_container_name" "$image_id" bash)    
+    docker start "$dummy_container_id"
+    
+    cmd="python3 -m pip list --outdated --format=freeze | grep -v '^\-e' | cut -d = -f 1 | xargs -n 1 python3 -m pip install -U; python3 -m pip freeze > /tmp/requirements-freeze.txt;"
+    docker exec -ti "$dummy_container_id" /bin/bash -c "${cmd}"
+    docker cp "$dummy_container_id":/tmp/requirements-freeze.txt /tmp/requirements-freeze.txt
+    docker commit "$dummy_container_id" fcuvm
+    cp /tmp/requirements-freeze.txt $requirements_freeze_file_location
+    
+    docker stop "$dummy_container_id"
+    docker rm -f "$dummy_container_name"
+    rm -f /tmp/requirements-freeze.txt
+}
+
+# Looks for outdated python packages and prompts user if they want to
+# run a version upgrade for all outdated packages.
+# 
+# Args:
+#   $1  path to the {arch-based}-requirements-freeze file.
+# 
+upgrade_version() {
+    declare requirements_freeze_file_location="$1"
+    say "Checking for newer package versions..."     
+    
+    image_id=$(docker images | grep fcuvm | awk '{print $3}' | head -n 1)
+    echo $image_id
+    outdated_packages=($(docker run -it $image_id python3 -m pip list --outdated --format=freeze | grep -v '^\-e' | cut -d = -f 1))
+
+    if [ "${#outdated_packages[@]}" -ne 0 ]; then
+        say "List of outdated packages"
+        printf '%s\n' "${outdated_packages[@]}"
```

Comment:
There is a primitive that prints text on the terminal called `say`. Could it be used here?